docs/decisions/0002-lerna-monorepo.rst

```diff
@@ -0,0 +1,108 @@
+
+As such, we should pay extra attention to managing dependencies in each packages, making informed decisions about whether a dependency should be included in an individual package's package.json file or the package.json file at the root of the repository.
+
+================================
+Running CI for branches and pull requests
+================================
+
+When a branch is pushed or a pull request is created on Github, CI will continue to run tests, linting, and other necessary build steps to ensure changes did not cause regressions. The difference now is that we rely on Lerna to run commands across all packages that have changed since the last release. This will ensure when a change is made to a package, the CI will be sure to run any associated tests for that specific package, including any of its dependencies and dependents across the monorepo. Output from each command Lerna runs concurrently across all packages are streamed in each Github Action step.
+
+================================
+NPM releases with Lerna
+================================
+
+Publishing packages to NPM remains a job of Github Actions, however, releases will no longer happen using ``semantic-release`` directly. Instead, we will rely on the ``lerna publish`` command to determine which packages have changed since the last merged commit and release, and only publish those. 
+
+Lerna is configured to use independent versioning with conventional commits, as opposed to keeping all package versions in sync. This will allow us to publish packages independently of each other, rather than in lock-step like some JS monorepos do (e.g., Babel, React). The benefit of independent versioning is that consuming applications won't get erroneous upgrades when code unrelated to their use case is updated in the monorepo. For example, say the ``edx/frontend-app-learning`` micro-frontend consumes ``@edx/frontend-enterprise-utils``. Without independent versioning, if we released a new version of ``@edx/frontend-enterprise-catalog-search``, then ``edx/frontend-app-learning`` would have an upgrade available when nothing about the package changed.
+
+The image that follows is a diagram outlining the NPM publishing process. When a PR is merged, Github Actions will run the necessary Lerna commands to run all linting, tests, and builds for each package that changed since the last release, including dependencies and dependents. Each package, independently released, follows semantic versioning to determine the package version, creates and pushes Git tags, and then publishes to NPM.
+
+.. image:: https://user-images.githubusercontent.com/2828721/116780392-82947000-aa4a-11eb-82ce-819604de935f.png
+
+************
+Consequences
+************
+
+* The existing ``frontend-enterprise`` NPM package will be deprecrated in favor of the 3 new NPM packages outlined above. We will need to upgrade all uses of ``frontend-enterprise`` to import from the relevant individual NPM packages instead.
```

Comment:
What's the plan to let people know that they have to upgrade and push them into actually doing it?